lib/rspec/mocks/proxy.rb

```diff
@@ -407,23 +407,29 @@ class PartialClassDoubleProxy < PartialDoubleProxy
     class ProxyForNil < PartialDoubleProxy
       def initialize(order_group)
         @warn_about_expectations = true
+        @disallow_expectations_on_nil = false
```

Comment:
As this has no real ties on `nil` and is already on a class called `ProxyForNil`, this and all corresponding methods should really just be `@disallow_expectations`